### ISSUE TITLE
[Windows} Unheadered zipped issue.

### ISSUE THREAD

**Issue by jjfmz42015** [SourMesen/Mesen#805]
When loading clean unheadered aka headerless roms that are zipped mesen fails to execute rom. If it is unzipped loads fine. Possibly due to mesen using crc for lookup?

if so maybe its seeing zip crc rather than unzip then crc lookup?

a change in this behavior would be a plus.

**Comment by SourMesen** [comment]
If your files do not have a .nes extension, this is expected. Mesen only tries to load some files based on their extension when inside archives.

Generally speaking, unheadered roms for the NES are silly - they are missing vital information from the cartridge which forces emulators to rely on databases (and creating romhacks without headers, etc. becomes impossible).  I would recommend against using unheadered roms in general.

**Comment by jjfmz42015** [comment]
I see. So then in your opinion due to header GoodNES>no intro?

**Comment by jjfmz42015** [comment]
Also a good example mrs pac man, with the header missing plays on punes, nestopia ue and mesen fine, on fceux it mirrors incorrect and wont play fine unless you fix the header, since mesen seems to figure it out on its own why need header?

**Comment by SourMesen** [comment]
This isn't related to goodnes vs no-intro, and I don't have an opinion there.
There are no-intro sets being distributed by people without headers, and others have headers, too.

The only reason an emulator can load a rom with a header *at all* is because the emulator contains a huge database of thousands of known roms with their correct headers.  Without this external database, headerless roms are impossible to load. Whereas headered roms contain all the information required for an emulator to load the rom, without requiring any external database.

The 16-byte headers in .nes files contain vital information about the cartridge - without it, there is no way for an emulator to know how to emulate the game.

**Comment by jjfmz42015** [comment]
True no into sets are headerless proper goodnes set is headered thats why i said it that way. Thanks for input

**Comment by Cibomatto2002** [comment]
Why do they make headerless roms?

**Comment by jjfmz42015** [comment]
> Why do they make headerless roms?

Rom headers are put there by the copier hardware or a tool after to explain how to load the rom, true clean code would be headerless, but without a data base in the emulator useless.

Mesen has a database,

**Comment by SourMesen** [comment]
One reason rom headers have a bad reputation is because on the SNES, the headers added by copiers were useless (for anything aside copiers.)

Additionally, SNES roms all contain a "header" giving information about the game, in the game's code itself (SNES developers were required to fill out this information), which is what is used to determine how to emulate SNES games.  An equivalent does not exist for the NES, so NES emulators had to come up with their own way to specify information about the cartridge - this is what the 16-byte .nes header does.

Basically:
![image](https://user-images.githubusercontent.com/21093973/79774881-6a4f2880-8301-11ea-957f-cfed4b68a885.png)

An unheadered ROM is only partial information about the cartridge, and is missing everything outside the red section in this screenshot.

**Comment by jjfmz42015** [comment]
> One reason rom headers have a bad reputation is because on the SNES, the headers added by copiers were useless (for anything aside copiers.)
> 
> Additionally, SNES roms all contain a "header" giving information about the game, in the game's code itself (SNES developers were required to fill out this information), which is what is used to determine how to emulate SNES games. An equivalent does not exist for the NES, so NES emulators had to come up with their own way to specify information about the cartridge - this is what the 16-byte .nes header does.
> 
> Basically:
> ![image](https://user-images.githubusercontent.com/21093973/79774881-6a4f2880-8301-11ea-957f-cfed4b68a885.png)
> 
> An unheadered ROM is only partial information about the cartridge, and is missing everything outside the red section in this screenshot.

What tool would you suggest for fixing headers on a NES dump? I found a few but am curious what you would recommend. Thanks

**Comment by Cibomatto2002** [comment]
Thank you for explaining it to me.

**Comment by jjfmz42015** [comment]
> I've never used one, so I can't help you there - odds are most of them aren't really up-to-date or perfectly accurate, either.
> 
> A very good database of information was recently posted here: http://forums.nesdev.com/viewtopic.php?f=3&t=19940
> But you'd have to write your own tool to apply them to roms.

One last question. Do FDS need headers as well? I assume not since they are floppy dumps but Ide like to know from someone with your knowledge.

**Comment by SourMesen** [comment]
No, headers on FDS files aren't useful (Mesen ignores them), because a FDS file by itself means you're emulating a very specific hardware (the FDS), so there is no need for additional information (the specs of the FDS do not vary, and different FDS games do not have different hardware specs)
